pylops/signalprocessing/dct.py

```diff
@@ -0,0 +1,96 @@
+__all__ = ["DCT"]
+
+from typing import List, Optional, Union
+
+import numpy as np
+from scipy import fft
+
+from pylops import LinearOperator
+from pylops.utils._internal import _value_or_sized_to_tuple
+from pylops.utils.decorators import reshaped
+from pylops.utils.typing import DTypeLike, InputDimsLike, NDArray
+
+
+class DCT(LinearOperator):
+    r"""Discrete Cosine Transform.
+
+    Apply 1D or ND-Cosine Transform along one or more ``axes`` of a multi-dimensional
+    array of size ``dims``.
+
+    This operator is an overload of :func:`scipy.fft.dctn` in forward mode and :func:`scipy.fft.idctn`
+    in adjoint mode.
+
+    Parameters
+    ----------
+    dims : :obj:`int` or :obj:`tuple`
+        Number of samples for each dimension
+    type : :obj:`int`, optional
+        Type of DCT (see scipy's documentation for more details). Default type is 2.
+    axes : :obj:`int` or :obj:`list`, optional
+        Axes over which the DCT is computed. If ``None``, the transform is applied
+        over all axes.
+    workers :obj:`int`, optional
+        Maximum number of workers to use for parallel computation. If negative,
+        the value wraps around from os.cpu_count().
+    dtype : :obj:`str`, optional
+        Type of elements in input array.
+    name : :obj:`str`, optional
+        Name of operator (to be used by :func:`pylops.utils.describe.describe`)
+
+    Attributes
+    ----------
+    shape : :obj:`tuple`
+        Operator shape
+    explicit : :obj:`bool`
+        Operator contains a matrix that can be solved explicitly
+        (``True``) or not (``False``)
```

Comment:
I would probably remove this. These are standard LinearOperator attributes.

Reply:
@cako this is something we have in all operators... I have been wondering for a while if it makes sense to remove them (or even better, remove them and update Attributes fields to contain special attributes of a given class which users may want to know about - eg for `Kirchhoff` variables like `trav`, `amp`, `cosangle`...

@dikwickley: this is not a comment for this PR, but if we agree I can make a GitHub issue to be tackled separately :)

Reply:
Ok, let's tackle this separately!